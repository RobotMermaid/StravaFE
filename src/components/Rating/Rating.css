.Rating {
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  color: rgb(16, 147, 27);
}
h1 {
  font-size: 2rem;
}
h2 {
  font-size: 1.5rem;
}
.root {
  width: 200;
  display: 'flex';
  align-items: 'center';

}
.stars {
  color: rgb(255, 60, 0);
}
.rated {
  visibility: visible;
}
.notRated {
  visibility: hidden;
}
.button {
  color: rgb(250, 248, 248);
  cursor: pointer;
  outline: rgb(255, 60, 0);
  background-color: rgb(255, 60, 0);
  border-radius: 0.5rem;
  border-color: white;
  margin-top: 1rem;
  padding: 0.25rem 0.3rem 0.25rem 0.3rem;
}
.button:hover {
  background-color: rgb(8, 8, 8);

}