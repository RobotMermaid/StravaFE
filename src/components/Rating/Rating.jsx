import React, { useState } from 'react';
import styles from './Rating.css';
import RatingMU from '@material-ui/lab/Rating';
import Box from '@material-ui/core/Box';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import { rateShoe } from '../Services/API';
import findShoeImg from '../Services/shoeImg';

const Rating = () => {
  const [athlete_id, setAthleteID] = useState(0);
  const [shoes_id, setshoes_id] = useState(0);
  const [anchorEl, setAnchorEl] = useState(0);
  const [selectedIndex, setSelectedIndex] = useState(1);

  const [stars, setStars] = useState(2);
  const [hover, setHover] = useState(-1);
  const [rated, setRated] = useState(false);
  
  let img;
  
  const options = [
    '',
    'Nike Zoom',
    'Altra Lone Peak',
    'Brooks Ghost',
    'Salomon Supercross',
    'Brooks Cascadia',
    'Salomon Ultra Pro',
    'Nike Pegasus'
  ];
  const handleClickListItem = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleMenuItemClick = (event, index) => {
    setshoes_id(index)
    setSelectedIndex(index);
    setAnchorEl(null);
  };
  img = findShoeImg(options[selectedIndex]);

  const handleClose = () => {
    setAnchorEl(null);
  };

  const saveRating = () => {
    console.log({athlete_id, shoes_id, stars})
    let athletes_id = parseInt(athlete_id)
    rateShoe({athletes_id, shoes_id, stars})
    setRated(!rated)
  }
  const labels = {
    0.5: 'Useless',
    1: 'Painful',
    1.5: 'Disgusting',
    2: 'You call this a shoe',
    2.5: 'Barefoot running is better',
    3: 'Meh',
    3.5: 'Good',
    4: 'Better',
    4.5: 'Excellent',
    5: 'Like running on clouds',
  };
  return (
    <div className={styles.Rating}>
      <h1 className={styles.h1}>Rate a shoe</h1>
      <p>For testing use Athlete Id of 1, 2 or 3</p>
      <form>
      <div>
        <p>Enter Athlete Id</p>
        <input data-testid="athleteID"
        text="athleteID"
        type="text" value={athlete_id} onChange={({ target }) => setAthleteID(target.value)} placeholder="Your Athlete ID"/>
      </div>
      <div className={styles.inputs}>
      <img src={img} />
        <List component="nav" aria-label="Shoes">
        <ListItem
          button
          aria-haspopup="true"
          aria-controls="lock-menu"
          aria-label="when device is locked"
          onClick={handleClickListItem}
        >
          <ListItemText primary="Select a shoe to rate" secondary={options[selectedIndex]} />
        </ListItem>
      </List>
      <Menu
        id="lock-menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleClose}
      >
        {options.map((option, index) => (
          <MenuItem
            key={option}
            disabled={index === 0}
            selected={index === selectedIndex}
            onClick={(event) => handleMenuItemClick(event, index)}
          >
            {option}
          </MenuItem>
        ))}
      </Menu>
      </div>
      <div className={styles.root}>
      <RatingMU
        className={styles.stars}
        name="hover-feedback"
        value={stars}
        precision={0.5}
        onChange={(event, newStars) => {
          setStars(newStars);
        }}
        onChangeActive={(event, newHover) => {
          setHover(newHover);
        }}
      />
      {stars !== null && <Box ml={2}>{labels[hover !== -1 ? hover : stars]}</Box>}
    </div>
    </form>
    <button className={styles.button} onClick={(e) => {e.preventDefault(); saveRating()}}>Submit your rating</button>
      <h3 className={rated ? styles.rated : styles.notRated }>Thank you for rating {options[selectedIndex]}</h3>
    </div>
  );
};

export default Rating;