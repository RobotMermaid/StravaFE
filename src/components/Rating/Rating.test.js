import React from 'react';
import { render, screen, fireEvent, waitFor, cleanup } from '@testing-library/react';
import Rating from './Rating';

describe('Rating component', () => {
  afterEach(() => cleanup());
  it('renders Rating', () => {
    const { asFragment } = render(<Rating />);
    expect(asFragment()).toMatchSnapshot();
  });
  
it('changes the athlete id text when the text input is changed', async() => {
  render(<Rating />);
  // const textInput = await screen.findByLabelText('Text');
  const textInput = await screen.findByPlaceholderText('Your Athlete ID');

  fireEvent.change(textInput, { target: { value: 1 } });
  await waitFor(() => {
    expect(textInput.value).toBe('1');
  });
});
});
