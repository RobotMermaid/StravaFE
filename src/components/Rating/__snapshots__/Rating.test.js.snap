// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component renders Rating 1`] = `
<DocumentFragment>
  <div
    class="Rating"
  >
    <h1
      class="h1"
    >
      Rate a shoe
    </h1>
    <p>
      For testing use Athlete Id of 1, 2 or 3
    </p>
    <form>
      <div>
        <p>
          Enter Athlete Id
        </p>
        <input
          data-testid="athleteID"
          placeholder="Your Athlete ID"
          text="athleteID"
          type="text"
          value="0"
        />
      </div>
      <div
        class="inputs"
      >
        <img
          src="/path/image"
        />
        <nav
          aria-label="Shoes"
          class="MuiList-root MuiList-padding"
        >
          <div
            aria-controls="lock-menu"
            aria-disabled="false"
            aria-haspopup="true"
            aria-label="when device is locked"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Select a shoe to rate
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                Altra Lone Peak
              </p>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </nav>
      </div>
      <div
        class="root"
      >
        <span
          class="MuiRating-root stars"
        >
          <span
            class="MuiRating-decimal"
          >
            <label
              class="MuiRating-label"
              for="hover-feedback-0-5"
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyhidden"
              >
                0.5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyhidden"
              id="hover-feedback-0-5"
              name="hover-feedback"
              type="radio"
              value="0.5"
            />
            <label
              class="MuiRating-label"
              for="hover-feedback-1"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyhidden"
              >
                1 Star
              </span>
            </label>
            <input
              class="MuiRating-visuallyhidden"
              id="hover-feedback-1"
              name="hover-feedback"
              type="radio"
              value="1"
            />
          </span>
          <span
            class="MuiRating-decimal"
          >
            <label
              class="MuiRating-label"
              for="hover-feedback-1-5"
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyhidden"
              >
                1.5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyhidden"
              id="hover-feedback-1-5"
              name="hover-feedback"
              type="radio"
              value="1.5"
            />
            <label
              class="MuiRating-label"
              for="hover-feedback-2"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyhidden"
              >
                2 Stars
              </span>
            </label>
            <input
              checked=""
              class="MuiRating-visuallyhidden"
              id="hover-feedback-2"
              name="hover-feedback"
              type="radio"
              value="2"
            />
          </span>
          <span
            class="MuiRating-decimal"
          >
            <label
              class="MuiRating-label"
              for="hover-feedback-2-5"
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyhidden"
              >
                2.5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyhidden"
              id="hover-feedback-2-5"
              name="hover-feedback"
              type="radio"
              value="2.5"
            />
            <label
              class="MuiRating-label"
              for="hover-feedback-3"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyhidden"
              >
                3 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyhidden"
              id="hover-feedback-3"
              name="hover-feedback"
              type="radio"
              value="3"
            />
          </span>
          <span
            class="MuiRating-decimal"
          >
            <label
              class="MuiRating-label"
              for="hover-feedback-3-5"
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyhidden"
              >
                3.5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyhidden"
              id="hover-feedback-3-5"
              name="hover-feedback"
              type="radio"
              value="3.5"
            />
            <label
              class="MuiRating-label"
              for="hover-feedback-4"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyhidden"
              >
                4 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyhidden"
              id="hover-feedback-4"
              name="hover-feedback"
              type="radio"
              value="4"
            />
          </span>
          <span
            class="MuiRating-decimal"
          >
            <label
              class="MuiRating-label"
              for="hover-feedback-4-5"
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyhidden"
              >
                4.5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyhidden"
              id="hover-feedback-4-5"
              name="hover-feedback"
              type="radio"
              value="4.5"
            />
            <label
              class="MuiRating-label"
              for="hover-feedback-5"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyhidden"
              >
                5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyhidden"
              id="hover-feedback-5"
              name="hover-feedback"
              type="radio"
              value="5"
            />
          </span>
        </span>
        <div
          class="MuiBox-root MuiBox-root-1"
        >
          You call this a shoe
        </div>
      </div>
    </form>
    <button
      class="button"
    >
      Submit your rating
    </button>
    <h3
      class="notRated"
    >
      Thank you for rating Altra Lone Peak
    </h3>
  </div>
</DocumentFragment>
`;
